### PR TITLE
Allocation Vaccine - New API Statistics & Changes at API List

### PR DESCRIPTION
**Pull request by rindibudiaramdhan**
## Changes
- API Statistic of Allocation Request
  - new API to get statistic of "alkes" allocation request
  - this API will give response of total all status requests, total draft requests and total success requests
  - implement local scope
- API Statistic of Allocation Vaccine Request
  - new API to get statistic of Allocation Vaccine Request
  - this API will give response of total all status requests, total draft requests and total success requests
  - implement local scope
- API Allocation Requests List
  - remove detail distributions and detail materials
  - implement local scope
- API Allocation Vaccine Requests List
  - remove detail distributions and detail materials
  - implement local scope
  
## References
- https://laravel.com/docs/8.x/eloquent#local-scopes

## Evidence
title: Allocation Vaccine - New API Statistics & Changes at API List
participants: @budiaramdhanrindi @khihadysucahyo @sandisunandar @firmanalamsyah580
project: Pikobar Logistik

### CODE REVIEW

**Review by ayocodingit — APPROVED** [review]

File: app/Http/Controllers/API/v1/AllocationRequestController.php [hunk 1]
```diff
@@ -12,17 +14,25 @@ class AllocationRequestController extends Controller
...
         ->find($id);
         return response()->format(Response::HTTP_OK, 'success', $data);
     }
+
+    public function statistic(Request $request)
+    {
+        return [
+            'total_requests' => AllocationRequest::where('type', AllocationRequestTypeEnum::alkes())->count(),
+            'total_draft' => AllocationRequest::where('type', AllocationRequestTypeEnum::alkes())->where('status', AllocationRequestStatusEnum::draft())->count(),
+            'total_success' => AllocationRequest::where('type', AllocationRequestTypeEnum::alkes())->where('status', AllocationRequestStatusEnum::success())->count(),
```
Comment (khihadysucahyo) [inline app/Http/Controllers/API/v1/AllocationRequestController.php:35]: is it possible if the code below is wrapped in one variable, it looks like this code looks repetitive(?)
`$alkesRequest = AllocationRequest::where('type', AllocationRequestTypeEnum::alkes())`

so just add an addtional query like this for example, `$alkesRequest->count()`, etc

Comment (khihadysucahyo) [reply app/Http/Controllers/API/v1/AllocationRequestController.php:35]: just spamming on this PR dude, sorry :D
or maybe u can use local scope https://laravel.com/docs/8.x/eloquent#local-scopes to handle this repetitive code:
`AllocationRequest::where('type', AllocationRequestTypeEnum::alkes())`

Comment (khihadysucahyo) [reply app/Http/Controllers/API/v1/AllocationRequestController.php:35]: just spamming on this PR dude, sorry :D
or maybe u can use local scope https://laravel.com/docs/8.x/eloquent#local-scopes to handle this repetitive code: `AllocationRequest::where('type', AllocationRequestTypeEnum::alkes())` 
This code is also seen used in the index method 

maybe u can considering to use local scope eg. `scopeAlkes()` , `scopeVaccine`

**Review by sandisunandar99 — APPROVED** [review]
L.N.J.T

File: app/Http/Controllers/API/v1/AllocationRequestController.php (hunk 1 above)
Comment (rindibudiaramdhan) [reply app/Http/Controllers/API/v1/AllocationRequestController.php]: done at https://github.com/jabardigitalservice/pikobar-logistik-api/pull/322/commits/8cc2a303f0ae55cd88ab9e0161456a43dd65193a & https://github.com/jabardigitalservice/pikobar-logistik-api/pull/322/commits/f78de7f59a1d62fb090b3f45ff1da55de20e1784 commits

**Review by khihadysucahyo — APPROVED** [review]

**Pull request merged**